cmd/run.go

```diff
@@ -270,8 +275,18 @@ a commandline interface for interacting with it.`,
 
 			// Start the test run
 			initBar.Modify(pb.WithConstProgress(0, "Starting test..."))
-			if err := engineRun(); err != nil {
-				return errext.WithExitCodeIfNone(err, exitcodes.GenericEngine)
+			var interrupt error
+			err = engineRun()
+			if err != nil {
+				if common.IsInterruptError(err) {
+					interrupt = err
+				}
+				if !conf.Linger.Bool {
+					if interrupt == nil {
+						return errext.WithExitCodeIfNone(err, exitcodes.GenericEngine)
+					}
+					return errext.WithExitCodeIfNone(interrupt, exitcodes.ScriptException)
```

Comment:
The current behavior (and in #1920) is to exit the script immediately without printing the test summary. Shouldn't we do a clean exit and print the test summary instead? This is already the behavior when `--linger` is used, so it would be consistent.

Reply:
Hm ... I guess it's better if we are consistent and it seems to just drop this return ? (and refactor the if's into one I guess)

Reply:
I'm weary of doing changes in `cmd`, so hopefully 2381b70c didn't break anything. Let me know.

The only slight nitpick would be that the error is shown _after_ the summary and not before, but I think we can live with that.